pkg/core/writer/writer.go

```diff
@@ -135,72 +144,89 @@ func New(conf *config.WriterConfig, dpChan chan []*datapoint.Datapoint, eventCha
+				Timeout:   3 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: conf.MaxRequests,
+			IdleConnTimeout:     30 * time.Second,
+			TLSHandshakeTimeout: 10 * time.Second,
+		}
 
-	eventEndpointURL := conf.ParsedEventEndpointURL()
-	if eventEndpointURL == nil {
-		var err error
-		eventEndpointURL, err = conf.ParsedIngestURL().Parse("v2/event")
+		dpEndpointURL, err := conf.ParsedIngestURL().Parse("v2/datapoint")
 		if err != nil {
 			log.WithFields(log.Fields{
 				"error":     err,
 				"ingestURL": conf.ParsedIngestURL().String(),
-			}).Error("Could not construct event ingest URL")
+			}).Error("Could not construct datapoint ingest URL")
 			return nil, err
 		}
+		sw.client.DatapointEndpoint = dpEndpointURL.String()
+
+		eventEndpointURL := conf.ParsedEventEndpointURL()
+		if eventEndpointURL == nil {
+			var err error
+			eventEndpointURL, err = conf.ParsedIngestURL().Parse("v2/event")
+			if err != nil {
+				log.WithFields(log.Fields{
+					"error":     err,
+					"ingestURL": conf.ParsedIngestURL().String(),
+				}).Error("Could not construct event ingest URL")
+				return nil, err
+			}
+		}
+		sw.client.EventEndpoint = eventEndpointURL.String()
+
+		traceEndpointURL := conf.ParsedTraceEndpointURL()
+		if traceEndpointURL == nil {
+			var err error
+			traceEndpointURL, err = conf.ParsedIngestURL().Parse(conf.DefaultTraceEndpointPath())
+			if err != nil {
+				log.WithFields(log.Fields{
+					"error":     err,
+					"ingestURL": conf.ParsedIngestURL().String(),
+				}).Error("Could not construct trace ingest URL")
+				return nil, err
+			}
+		}
+		sw.client.TraceEndpoint = traceEndpointURL.String()
+
+		// The only reason this is on the struct and not a local var is so we can
+		// easily get diagnostic metrics from it
+		sw.serviceTracker = sw.startHostCorrelationTracking()
 	}
-	sw.client.EventEndpoint = eventEndpointURL.String()
 
-	traceEndpointURL := conf.ParsedTraceEndpointURL()
-	if traceEndpointURL == nil {
-		var err error
-		traceEndpointURL, err = conf.ParsedIngestURL().Parse(conf.DefaultTraceEndpointPath())
+	go sw.maintainLastMinuteActivity()
+
+	if conf.Splunk.Enabled {
+		writer, err := splunk.Build(conf.Splunk.URL, conf.Splunk.Token, conf.Splunk.Source, conf.Splunk.SourceType, conf.Splunk.Index, conf.Splunk.SkipTLSVerify, conf.Splunk.Host)
+
 		if err != nil {
-			log.WithFields(log.Fields{
-				"error":     err,
-				"ingestURL": conf.ParsedIngestURL().String(),
-			}).Error("Could not construct trace ingest URL")
 			return nil, err
 		}
+		sw.splunkWriter = &writer
+		log.Infof("Sending splunk data to %s", conf.Splunk.URL)
 	}
-	sw.client.TraceEndpoint = traceEndpointURL.String()
 
 	sw.datapointFilters, err = sw.conf.DatapointFilters()
 	if err != nil {
 		return nil, err
 	}
 
-	sw.dimensionClient.Start()
-	sw.correlationClient.Start()
-
-	go sw.listenForEventsAndDimensionUpdates()
+	if conf.SignalFxAccessToken != "" {
```

Comment:
Change this conditional to be based on whether the sub clients are nil or not.

Reply:
Done.